libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -75,6 +77,126 @@ impl Subpath {
 			(Subpath::new(first_split, false), Some(Subpath::new(second_split, false)))
 		}
 	}
+
+	// Helper function to clip overlap of two intersecting open Subpaths. Returns an optional, as occasionally the approximation for finding intersections
+	// doesn't return intersections at the very endpoints if it falls within the error threshold.
+	// TODO: If a segment curls back on itself tightly enough it could intersect again at the portion that would be trimmed. This could cause the subpath
+	//  subpaths to clip at the incorrect location. This can be avoided by first trimming the two subpaths at any extrema, effectively ignoring loopbacks.
+	fn clip_subpaths(subpath1: &Subpath, subpath2: &Subpath) -> Option<(Subpath, Subpath)> {
+		// Split the first subpath at its last intersection
+		let intersections1 = subpath1.subpath_intersections(subpath2, None, None);
+		if intersections1.is_empty() {
+			return None;
+		}
+		let last_intersection = *intersections1.last().unwrap();
+		let (clipped_subpath1, _) = subpath1.split(SubpathTValue::GlobalParametric(last_intersection));
+
+		// Split the second subpath at its first intersection
+		let intersections2 = subpath2.subpath_intersections(subpath1, None, None);
+		if intersections2.is_empty() {
+			return None;
+		}
+		let first_intersection = intersections2[0];
+		let (_, clipped_subpath2) = subpath2.split(SubpathTValue::GlobalParametric(first_intersection));
+
+		Some((clipped_subpath1, clipped_subpath2.unwrap()))
+	}
+
+	/// Reduces the segments of the subpath into simple subcurves, then scales each subcurve a set `distance` away.
+	/// The intersections of segments of the subpath are joined using the method specified by the `joint` argument.
+	/// <iframe frameBorder="0" width="100%" height="375px" src="https://graphite.rs/bezier-rs-demos#subpath/offset/solo" title="Offset Demo"></iframe>
+	pub fn offset(&self, distance: f64, joint: Joint) -> Subpath {
+		assert!(self.len_segments() > 1, "Cannot outline an empty Subpath.");
+
+		let mut subpaths = self.iter().map(|bezier| Subpath::from_beziers(bezier.offset(distance), false)).collect::<Vec<Subpath>>();
+		let mut drop_common_point = vec![true; self.len()];
+
+		if distance != 0. {
+			// Clip or joining consecutive Subpaths
+			for i in 0..subpaths.len() - 1 {
+				let j = i + 1;
+				let subpath1 = &subpaths[i];
+				let subpath2 = &subpaths[j];
+
+				let last_segment = subpath1.get_segment(subpath1.len_segments() - 1).unwrap();
```

Comment:
Just want to confirm: We're sure these subpaths won't be empty right?

Reply:
I don't believe its possible for the subpaths to be empty